keps/sig-auth/3221-multiple-authorization-webhooks/README.md

```diff
@@ -0,0 +1,556 @@
+However since `kube-apiserver` only allows specifying a single webhook, this would result in cluster administrators being not able to specify their own.
+Support for multiple webhooks would make this possible.
+Moreover, RBAC rules can't be used here since RBAC allows to one add permissions but not deny them.
+We won't be able to add restrictions on 'non-system' users only for a set of resources using RBAC; 'non-system' users here refers to anyone who shouldn't be able to update/delete the protected set of CRDs.
+
+Relevant Discussion Thread is sig-apimachinery: https://groups.google.com/g/kubernetes-sig-api-machinery/c/MBa19WTETMQ
+
+#### Story 2: Preventing unnecessarily nested webhooks
+
+Consider a system administrator who would like to apply a set of validations to certain requests before handing it off to webhooks defined using frameworks like Open Policy Agent.
+
+They would have to nested webhooks within the one added to the auth chain to have the intended effect.
+This enhancement allows the administrator to configure this behavior via a structured API and invoke the additional webhook only when relevant.
+This also allows administrators to define `onError` behavior for separate webhooks, leading to more predictable outcomes.
+
+#### Story 3: Denying requests on certain scenarios
+
+The authorizer chain should be powerful enough to deny anyone making a request if certain conditions are satisfied.
+
+#### Story 4: Varying defaults across versions of the API
+
+If the default values for configuration evolve over time, affected users might have to override these values in case they are affected by supplying the flags with updated defaults.
+A configuration format like the one proposed can be versioned over time with changing defaults, mitigating this risk by only affecting users who knowingly bump the version.
+
+### Notes/Constraints/Caveats (Optional)
+
+TBD
+
+### Risks and Mitigations
+
+- In HA clusters, there may be a skew in how the `kube-apiserver` processes in each are configured. This may create inconsistencies. Mitigation is to have the cluster bootstrapper handle such scenarios.
+- In case an administrator enables this feature and the webhook kubeconfig file is invalid or doesn't exist at the specified path, `kube-apiserver` on that node will not be able to start. This can be mitigated by fixing the malformed values.
+
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+We would like to introduce a structured file format which allows authorization to be configured using a flag (`--authorization-config-file`) which accepts a path to a file on the disk.
+This feature can be enabled or disabled by the explicit feature flag `--authorization-config-from-file`.
```

Comment:
What happens if an admin tries to use both the old and new way at the same time?